src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -838,12 +823,13 @@ public bool IsDaylightSavingTime()
 
         public static DateTime SpecifyKind(DateTime value, DateTimeKind kind)
         {
-            return new DateTime(value.InternalTicks, kind);
+            if ((uint)kind > (uint)DateTimeKind.Local) ThrowInvalidKind();
+            return new DateTime(value.UTicks | ((ulong)kind << KindShift));
         }
 
         public long ToBinary()
         {
-            if (Kind == DateTimeKind.Local)
+            if ((_dateData & LocalMask) != 0)
```

Comment:
>LocalMask [](start = 29, length = 9)

could you please add a comment here telling we are checking for Local Kind?

Reply:
Removed the `LocalMask` constant entirely and switched to using `KindLocal` everywhere - this should help with clarity.